src/compression.jl

```diff
@@ -96,12 +86,22 @@ function TranscodingStreams.finalize(codec::ZstdCompressor)
         end
         codec.cstream.ptr = C_NULL
     end
-    reset!(codec.cstream.ibuffer)
-    reset!(codec.cstream.obuffer)
-    return
+    nothing
 end
 
 function TranscodingStreams.startproc(codec::ZstdCompressor, mode::Symbol, error::Error)
+    if codec.cstream.ptr == C_NULL
+        ptr = LibZstd.ZSTD_createCStream()
+        if ptr == C_NULL
+            throw(OutOfMemoryError())
+        end
+        codec.cstream.ptr = ptr
```

Comment:
```suggestion
        codec.cstream.ptr = LibZstd.ZSTD_createCStream()
        if codec.cstream.ptr == C_NULL
            throw(OutOfMemoryError())
        end
```

There is no point in using a temporary variable here. `codec.cstream.ptr` starts as `C_NULL` anyways.